processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -666,6 +666,30 @@ func removeUnnecessaryPodData(pod *api_v1.Pod, rules ExtractionRules) *api_v1.Po
 	return &transformedPod
 }
 
+// parseAttributesFromImage parses the image name and tag for differently-formatted image names.
+// returns "latest" as the default if tag not present. also checks if the image contains a digest.
+// if it does, no latest tag is assumed.
+func parseNameAndTagFromImage(image string) (name, tag string, err error) {
```

Comment:
Should we use the already available internal [docker.ParseImageName](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/common/docker/images.go#L23) function instead?

I would prefer to all components use the same strategy for common functionalities. Note that the shared function cannot parse images with digest at the moment and should be fixed beforehand: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36279

Reply:
Having a single implementation makes sense.
Also I think the package should be renamed to `container` and not `docker` since it doesn't only cover docker.

Reply:
@spiffyy99 I was gonna merge this before the release but now I want to make our test coverage is still handling when no digest is set. If you get to using `docker.ParseImageName` before this is merged I'll close the issue.

Reply:
sure, I can update.

Reply:
actually, looking at this further, wouldn't it be better to use the official docker api code for things like this? i'm sure this regex is correct as well but I'd think that the official one is a better practice (reference.Parse).

another option is we can update `docker.ParseImageName` to use `reference.Parse`, but it seems like that function is used in a bunch

i'm not the one maintaining this though, so I don't feel strongly. just pointing it out